vendor/github.com/weaveworks/common/middleware/logging.go

```diff
@@ -82,6 +91,9 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 		statusCode, writeErr := wrapped.getStatusCode(), wrapped.getWriteError()
 
 		if writeErr != nil {
+			if errors.Is(writeErr, DoNotLogError{}) {
```

Comment:
My thinking was that you might already have an error wrapper, in which case you can implement `Is` on that and not be forced to add this extra wrapper.

Reply:
I based my implementation on the example here: 
https://github.com/golang/go/blob/089e37a931d30d4055c7468facb602c4cfa9b537/src/errors/wrap.go#L36C4-L39
which contradicts your idea of "this specific instance of error".

Reply:
That example overrides the method to check that the error is `fs.ErrExist` which is a specific instance. It also mentions `syscall.Errno.Is` as an example, which also checks equality to a given number of specific instances of `oserror` errors.

Reply:
(Note that neither of them just returns `true` because it's the same type. IMO, that's what `errors.As()` check is for.)

Reply:
I understand that the purpose of letting implementations provide their own `Is() bool` is to be able to return custom errors which behave like known errors. In that example you link, we want a custom error to be interpreted by downstream code as `fs.ErrExist`.

To follow this parttern: if this library had a `var ErrDoNotLog = errors.New("do not log this")` then we could check `errors.Is(err, ErrDoNotLog)`. In order to do that, we would move the implementation of `DoNotLogError` to Mimir codebase, and implement 
```go
func (DoNotLogError) Is(target err) bool { return target == middleware.ErrDoNotLog }
```

Reply:
I did that at first, but then I needed to do it again in weaveworks/common/middleware tests, and anyone else using the feature in a similar way would have to copy the code, so I just put it where it is.

Also I didn't bother exporting a value `ErrDoNotLog` because it worked ok using "any old value". 

While we're discussing it, my first idea was to check on an interface like `err.(interface { DoNotLog() bool })`, but you still need a type to pass to `errors.Is` so I dropped that idea too.

Reply:
Hence my suggestion which is to just:
- Drop the custom `Is()` implementation.
- Replace the `errors.Is(err, DoNotLog{})` call by `errors.As(err, &DoNotLog{})`.